lib/instrumentation/index.js

```diff
@@ -142,7 +150,9 @@ function Instrumentation(agent) {
       // Lazy require so that we don't have to use `require.resolve` which
       // would fail in combination with Webpack. For more info see:
       // https://github.com/elastic/apm-agent-nodejs/pull/957
-      return require(`./modules/${pathName}.js`)(...args);
+      const modExt = pathName.endsWith('.js') ? '' : '.js';
```

Comment:
Note for reviewer: now we can get a plain module name or a sub-module with the `.js` extension. The patch function needs to be aware of the extesion